File: src/easy/118-pascals-triangle/pascalsTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  const output = [];
  let i = 0;

  while (i < numRows) {
    const temporaryArray = new Array(i + 1).fill(1);
    
    if (i >= 2) {
      for (let j = 1; j < temporaryArray.length - 1; j++) {
        temporaryArray[j] = output[i - 1][j - 1] + output[i - 1][j];
      }
    }

    output.push(temporaryArray);

    i++;
  }

  return output;
};

console.log(generate(1)); // output == [[1]]
console.log(generate(5)); // output == [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]

File: src/easy/118-pascals-triangle/pascals_triangle.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        output = []
        i = 0

        while i < numRows:
            arr = []

            for j in range(i + 1):
                if (i > 0 and j > 0 and j < i):
                    k = output[i - 1][j - 1] + output[i - 1][j]
                    arr.append(k)
                else:
                    arr.append(1)

            output.append(arr)
            i += 1

        return output


testSolution = Solution()
print(testSolution.generate(1))  # output == [[1]]
print(testSolution.generate(5))  # output == [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]
